graphiql/src/App.tsx

```diff
@@ -135,6 +145,28 @@ function Core(props: { schema: GraphQLSchema | null }) {
   );
 }
 
+function computeHeadersString(
+  originalHeaders: string,
+  token: string | null
+): string {
+  if (!token) {
+    return originalHeaders;
+  }
+
+  try {
+    const headersJson = originalHeaders ? JSON.parse(originalHeaders) : {};
+    headersJson["Authorization"] = `Bearer ${token}`;
+    // If the headersJson is empty, we return an empty string to avoid GraphiQL to display {} in the
+    // headers editor
```

Comment:
Should probably be: "If headersJson is empty, we return an empty string to avoid GraphiQL displaying {} in the headers editor"